acceptance/framework/k8s/deploy.go

```diff
@@ -72,6 +73,33 @@ func DeployKustomize(t *testing.T, options *k8s.KubectlOptions, noCleanupOnFailu
 	RunKubectl(t, options, "wait", "--for=condition=available", "--timeout=5m", fmt.Sprintf("deploy/%s", deployment.Name))
 }
 
+func DeployJob(t *testing.T, options *k8s.KubectlOptions, noCleanupOnFailure bool, noCleanup bool, debugDirectory, kustomizeDir string) {
+	t.Helper()
+
+	KubectlApplyK(t, options, kustomizeDir)
+
+	output, err := RunKubectlAndGetOutputE(t, options, "kustomize", kustomizeDir)
+	require.NoError(t, err)
+
+	job := batchv1.Job{}
+	err = yaml.NewYAMLOrJSONDecoder(strings.NewReader(output), 1024).Decode(&job)
+	require.NoError(t, err)
+
+	helpers.Cleanup(t, noCleanupOnFailure, noCleanup, func() {
+		// Note: this delete command won't wait for pods to be fully terminated.
+		// This shouldn't cause any test pollution because the underlying
+		// objects are deployments, and so when other tests create these
+		// they should have different pod names.
+		WritePodsDebugInfoIfFailed(t, options, debugDirectory, labelMapToString(job.GetLabels()))
+		KubectlDeleteK(t, options, kustomizeDir)
+	})
+	logger.Log(t, "job deployed")
+
+	// The timeout to allow for connect-init to wait for services to be registered by the endpoints controller.
+	//RunKubectl(t, options, "wait", "--for=condition=", "--timeout=5m", fmt.Sprintf("job/%s", job.Name))
+	time.Sleep(30 * time.Second)
```

Comment:
Great call out. We can replace this with a wait that looks for the ready status of Pods created by the job:

```suggestion
	// Because Jobs don't have a "started" condition, we have to check the status of the Pods they create.
	RunKubectl(t, options, "wait", "--for=condition=Ready", "--timeout=5m", "pods", "--selector", fmt.Sprintf("job-name=%s", job.Name))

```